x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportExplainDataFrameAnalyticsAction.java

```diff
@@ -84,17 +100,37 @@ protected void doExecute(Task task,
 
     private void explain(Task task, PutDataFrameAnalyticsAction.Request request,
                          ActionListener<ExplainDataFrameAnalyticsAction.Response> listener) {
-        ExtractedFieldsDetectorFactory extractedFieldsDetectorFactory =
-                new ExtractedFieldsDetectorFactory(new ParentTaskAssigningClient(client, task.getParentTaskId()));
-        extractedFieldsDetectorFactory.createFromSource(
-            request.getConfig(),
-            ActionListener.wrap(
-                extractedFieldsDetector -> explain(task, request, extractedFieldsDetector, listener),
-                listener::onFailure)
+
+        final ExtractedFieldsDetectorFactory extractedFieldsDetectorFactory = new ExtractedFieldsDetectorFactory(
+            new ParentTaskAssigningClient(client, task.getParentTaskId())
         );
+        if (licenseState.isSecurityEnabled()) {
+            useSecondaryAuthIfAvailable(this.securityContext, () -> {
```

Comment:
`this.` is reduntant here AFAICT

Reply:
I like using `this`. It prevents the cognitive load of worrying about if the this is a member or a new variable.